gammapy/cube/fit.py

```diff
@@ -263,8 +263,15 @@ def _energy_axis(self):
 
     @property
     def data_shape(self):
-        """Shape of the counts data (tuple)"""
-        return self.counts.data.shape
```

Comment:
I just noticed we have hidden `MapDataset._geom` property, which already does exactly this. The data shape is also available as `geom.data_shape`. So my proposal would be to make the `._geom` public as and maybe call it `.geom_ref`. Then you can also use it in the `MapDatasetMaker`. The `ValueError` you can also move to `geom_ref`.

Reply:
Any reason why you want it to be called `geom_ref` instead of `geom`?